domain/service/json.go

```diff
@@ -15,10 +15,38 @@ package service
 
 import (
 	"encoding/json"
+
+	"github.com/control-center/serviced/domain"
 	"github.com/control-center/serviced/domain/servicedefinition"
 	"github.com/zenoss/glog"
 )
 
+type service Service
+
+func (s *Service) UnmarshalJSON(b []byte) error {
+	svc := service{}
+	if err := json.Unmarshal(b, &svc); err == nil {
+		*s = Service(svc)
+	} else {
+		return err
+	}
+	if len(s.Commands) > 0 {
+		s.Runs = nil
+		return nil
+	}
+	if len(s.Runs) > 0 {
+		s.Commands = make(map[string]domain.Command)
+		for k, v := range s.Runs {
+			s.Commands[k] = domain.Command{
+				Command:         v,
+				CommitOnSuccess: false,
```

Comment:
Ed brought this up, too. I was under the apparently mistaken impression that we wouldn't have a user on 5.0.x and 1.1, so maybe that's the way to go. I did bring this up with Ian, and his off-the-cuff answer was to default to false, but maybe we should run your point by him. I'll bring it up Monday morning.